ERCS/erc-7820.md

```diff
@@ -0,0 +1,411 @@
+
+The core of the standard includes:
+
+- **Registration and Unregistration**: Contracts can register with the ACR, specifying an admin who can manage roles within the contract. Contracts can also be unregistered when they are no longer active.
+
+- **Role Management**: Admins can grant or revoke roles for accounts, either individually or in batches, ensuring fine-grained control over who can perform what actions within a contract.
+
+- **Role Verification**: Any account can verify if another account has a specific role in a registered contract, providing transparency and facilitating easier integration with other systems.
+
+By centralizing access control management, the ACR standard aims to reduce redundancy, minimize errors in access control logic, and provide a clear and standardized approach to role management across smart contracts. This improves security and maintainability, making it easier for developers to implement robust access control mechanisms in their applications.
+
+## Motivation
+
+As decentralized applications (dApps) grow in complexity, managing access control across multiple smart contracts becomes increasingly difficult.
+Current practices involve bespoke implementations, leading to redundancy and potential security flaws. A standardized approach for managing roles and permissions will ensure better interoperability, security, and transparency. By providing a unified interface for registering contracts and managing roles, this standard simplifies development, ensures consistency, and enhances security. It facilitates easier integration and auditing, fostering a more robust and interoperable ecosystem.
+
+The advantages of using the provided system might be:
+
+Structured smart contracts management via specialized contracts.
+Ad-hoc access-control provision of a protocol.
+Ability to specify custom access control rules to maintain the protocol.
+
+## Specification
+
+The `AccessControlRegistry` contract provides a standardized interface for managing access control in Ethereum smart contracts. It includes functions to register and unregister contracts, grant and revoke roles for specific contracts, and check if an account has a particular role in a registered contract. Events are emitted for contract registration, unregistration, role grants, and role revocations, ensuring transparency and traceability of access control changes.
```

Comment:
Some of this is already stated in your abstract/motivation. That's not necessarily bad, but I just wanted to mention it.

> Events are emitted for contract registration, unregistration, role grants, and role revocations, ensuring transparency and traceability of access control changes.

You'll probably want to be a bit more specific about when each event type is fired.